providers/hexonet/records.go

```diff
@@ -182,7 +186,7 @@ func (n *HXClient) updateZoneBy(params map[string]string, domain string) error {
 func (n *HXClient) getRecords(domain string) ([]*HXRecord, error) {
 	var records []*HXRecord
 	zone := domain + "."
-	cmd := map[string]string{
+	cmd := map[string]interface{}{
 		"COMMAND":  "QueryDNSZoneRRList",
```

Comment:
Please be so kind as to use here also `RequestAllResponsePages` to query all data and loop over the responses.
Sorry for the circumstance and the effort you guys have. I am impressed how well and fast you went through the changes! Very very appreciated. I am unlucky that we have to enforce that way.

We are planning a new API behind the scenes that finally leads to less code in DNSControl and easier extensions if necessary. Finally you won't have to deal with commands, just a nice SDK communicating in json. It will take some time until we come there, but then! :-)

Reply:
Ok, I tried that but I have some bad news and some good news.

The bad news it that when I use `RequestAllResponsePages` in the same way that I used it for `GetZones()` I get this error:

```
    integration_test.go:173: runTests: Failed loading resource records for zone a-b-c-movies.com. [%!s(int=503) Invalid Parameter: FIRST]
```

Maybe I'm not doing it right?  I made nearly the exact same change as for GetZones.

The good news is that without any changes the code works for our integration test where we create/modify/delete zones of size 101, 601, and 1201.  The output of the tests is here: https://gist.github.com/tlimoncelli/a1869aee6a6f1e61379790473cded51d

I increased the testing to 9999 records and the entire create/modify/delete cycle worked just fine.  Maybe it is fine as-is?

Reply:
Ah, my fault. That's probably a wrong behavior/incorrect implementation of command `QueryDNSZoneRRList` in our API. In general, all our list query commands are limited or they should be.
So using `RequestAllResponsePages` is valid for the two use cases around the `QueryDNSZoneList` command, but not for the command `QueryDNSZoneRRList`. So, use the `Request` method for `QueryDNSZoneRRList` and the `RequestAllResponsePages` method for `QueryDNSZoneRRList`.
Why? Well, looks like `QueryDNSZoneRRList` is not limited and returns all records needed, but `QueryDNSZoneList` is limited to return 10000 zones at max.

Sorry for the confusion. HTH.

Reply:
I've tested it up to 15,000 records in a zone.  No problem so far.